docs/learn/async.md

```diff
@@ -72,112 +78,115 @@ class FormatBook(BaseTool):
         await asyncio.sleep(1)
         return f"{self.title} by {self.author}"
 
-
 @openai.call(model="gpt-4o-mini", tools=[FormatBook])
-@prompt_template("Recommend a {genre} book")
-async def recommend_book(genre: str):
-    ...
-
+@prompt_template()
+async def recommend_book_prompt(genre: str) -> Messages.Type:
+    return f"Recommend a {genre} book"
 
 async def main():
-    response = await recommend_book("fantasy")
+    response = await recommend_book_prompt("fantasy")
     if isinstance((tool := response.tool), FormatBook):
         output = await tool.call()
         print(output)
     else:
         print(response.content)
 
-
 asyncio.run(main())
 ```
 
-!!! tip "Type Hints"
+This example:
 
-    When using and calling asynchronous tools, you can check the type of the tool to get proper type hints.
+1. Defines an asynchronous tool `FormatBook` with an async `call` method.
+2. Uses this tool in an asynchronous OpenAI call.
+3. In the main function, it checks if the response includes a tool call, and if so, calls the tool asynchronously.
```

Comment:
we should mention that we check the instance type of the tool to ensure awaiting the tool call is correct